src/main/scala/util/ShiftReg.scala

```diff
@@ -84,11 +85,41 @@ object AsyncResetShiftReg {
     apply (in, depth, init.litValue.toInt, None)
 }
 
+class SynchronizerPrimitiveShiftRegVec(depth: Int = 1, init: Int = 0) extends AbstractPipelineReg(1) {
```

Comment:
why is this a `Vec` if it is only 1 bit wide? I think yuo should drop the `Vec` from the name